src/coreclr/jit/importer.cpp

```diff
@@ -4782,6 +4788,415 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
+
+            // ldarg.0
+            // conv.u
+            // ret
+            CLANG_FORMAT_COMMENT_ANCHOR;
+
+#ifdef TARGET_64BIT
+            bool uns = true;
+#else
+            bool uns                    = false;
+#endif
+
+            GenTree* op1 = impPopStack().val;
+            impBashVarAddrsToI(op1);
+
+            return gtNewCastNode(TYP_I_IMPL, op1, uns, TYP_U_IMPL);
+        }
+
+        case NI_SRCS_UNSAFE_AsRef:
+        {
+            assert(sig->sigInst.methInstCount == 1);
+
+            // ldarg.0
+            // ret
+
+            return impPopStack().val;
+        }
+
+        case NI_SRCS_UNSAFE_ByteOffset:
+        {
+            assert(sig->sigInst.methInstCount == 1);
+
+            // ldarg.1
+            // ldarg.0
+            // sub
+            // ret
+
+            GenTree* op2 = impPopStack().val;
+            GenTree* op1 = impPopStack().val;
+            impBashVarAddrsToI(op1, op2);
+
+            var_types type = impGetByRefResultType(GT_SUB, /* uns */ false, &op2, &op1);
+            return gtNewOperNode(GT_SUB, type, op2, op1);
+        }
+
+        case NI_SRCS_UNSAFE_Copy:
+        {
+            assert(sig->sigInst.methInstCount == 1);
+
+            // ldarg.0
+            // ldarg.1
+            // ldobj !!T
+            // stobj !!T
+            // ret
+
+            return nullptr;
+        }
+
+        case NI_SRCS_UNSAFE_CopyBlock:
+        {
+            assert(sig->sigInst.methInstCount == 0);
+
+            // ldarg.0
+            // ldarg.1
+            // ldarg.2
+            // cpblk
+            // ret
+
+            return nullptr;
+        }
+
+        case NI_SRCS_UNSAFE_CopyBlockUnaligned:
+        {
+            assert(sig->sigInst.methInstCount == 0);
+
+            // ldarg.0
+            // ldarg.1
+            // ldarg.2
+            // unaligned. 0x1
+            // cpblk
+            // ret
+
+            return nullptr;
+        }
+
+        case NI_SRCS_UNSAFE_InitBlock:
+        {
+            assert(sig->sigInst.methInstCount == 0);
+
+            // ldarg.0
+            // ldarg.1
+            // ldarg.2
+            // initblk
+            // ret
+
+            return nullptr;
+        }
+
+        case NI_SRCS_UNSAFE_InitBlockUnaligned:
+        {
+            assert(sig->sigInst.methInstCount == 0);
+
+            // ldarg.0
+            // ldarg.1
+            // ldarg.2
+            // unaligned. 0x1
+            // initblk
+            // ret
+
+            return nullptr;
+        }
+
+        case NI_SRCS_UNSAFE_IsAddressGreaterThan:
+        {
+            assert(sig->sigInst.methInstCount == 1);
+
+            // ldarg.0
+            // ldarg.1
+            // cgt.un
+            // ret
+
+            GenTree* op2 = impPopStack().val;
+            GenTree* op1 = impPopStack().val;
+
+            GenTree* tmp = gtNewOperNode(GT_GT, TYP_INT, op1, op2);
+            tmp->gtFlags |= GTF_RELOP_NAN_UN | GTF_UNSIGNED;
+            return gtFoldExpr(tmp);
```

Comment:
The reason we set it for `cgt.un` in the general case is because it can have FP operands.

I would prefer we do not make the code more confusing by having redundant flags.

Reply:
If its important, then we should likewise not set redundant flags for the regular import where that's relevant.

Unnecessarily deviating from the existing flags/semantics increases the scope and complexity of the PR. Handling it as a separate PR and fixing it more generally is its own issue.

Reply:
Removed the flag